src/pybind/ceph_argparse.py

```diff
@@ -442,7 +442,7 @@ class CephOsdName(CephArgtype):
 
     osd.<id>, or <id>, or *, where id is a base10 int
     """
-    def __init__(self):
+    def __init__(self, **kwargs):
```

Comment:
i don't understand this change, as `argdesc.__init__()` is the only place where we construct these `CephArgtype` instances. but since this method's signature looks like
```py
   def __init__(self, t, name=None, n=1, req=True, positional=True, **kwargs):
```

any unknown parameter should be consumed by `CephArgtype` already. actually after reverting this change and applying following patch
```diff
diff --git a/src/pybind/ceph_argparse.py b/src/pybind/ceph_argparse.py
index b8d4664cf8..8782b6a70a 100644
--- a/src/pybind/ceph_argparse.py
+++ b/src/pybind/ceph_argparse.py
@@ -509,7 +509,7 @@ class CephBool(CephArgtype):
     A boolean argument, values may be case insensitive 'true', 'false', '0',
     '1'.  In keyword form, value may be left off (implies true).
     """
-    def __init__(self, strings='', **kwargs):
+    def __init__(self, strings=''):
         self.strings = strings.split('|')

     def valid(self, s, partial=False):
```

i ran following tests:
```
$ bin/ceph config set global mon_warn_on_pool_pg_num_not_power_of_two false --force=false
$ bin/ceph config set global mon_warn_on_pool_pg_num_not_power_of_two true
$ bin/ceph orch device ls gnit json
Error ENOENT: No orchestrator configured (try `ceph orch set backend`)
```